packages/flutter/test/animation/animation_controller_test.dart

```diff
@@ -580,6 +580,93 @@ void main() {
     statusLog.clear();
   });
 
+  test('calling repeat with reverse set to true makes the animation alternate '
+      'between lowerBound and upperBound values on each repeat', () {
+    final AnimationController controller = AnimationController(
+      duration: const Duration(milliseconds: 100),
+      value: 0.0,
+      lowerBound: 0.0,
+      upperBound: 1.0,
+      vsync: const TestVSync(),
+    );
+
+    expect(controller.value, 0.0);
+
+    controller.repeat(reverse: true);
+    tick(const Duration(milliseconds: 0));
```

Comment:
It's not clear why this first duration=0 tick is always needed. I assume it's to enable the simulation to figure out which way it's going. Might be helpful to include one comment at the beginning here, to explain what's going on.

Reply:
This can also be found in other tests (e.g. `test('animateTo sets correct status')` or `test('after a reverse call animateTo sets correct status')`. Every time when starting a new simulation, the first tick is used to initialise the simulation inside the `AnimationController`. When the line is removed, the elapsed time will always be 0 which means that the simulation will be evaluated at the time 0 and not after x milliseconds. 

In the end I didn't comment on that because it is used quite often in the other tests already 😉

Reply:
Good point, OK.